File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/entity/RolePermissionGroupMapping.java
package com.nivasafinance.features.rolemanagement.mapping.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "n_role_permission_group_mapping")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class RolePermissionGroupMapping extends AuditableEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "role", nullable = false)
    private String role;
    
    @Column(name = "permission_group_id", nullable = false)
    private Long permissionGroupId;
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/entity/RolePermissionGroupMapping.kt
package com.nivasafinance.features.rolemanagement.mapping.entity

import com.nivasafinance.common.audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "role_permission_group_mapping")
class RolePermissionGroupMapping(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    val id: UUID? = null,

    @Column(name = "role_id", nullable = false)
    val roleId: UUID,

    @Column(name = "permission_group_id", nullable = false)
    val permissionGroupId: UUID
) : AuditableEntity()
